autoscheduler/frontend/src/tests/ui/SectionSelect.test.tsx

```diff
@@ -67,6 +69,124 @@ function makeCourseCard(...args: any): CourseCardOptions {
 }
 
 describe('SectionSelect', () => {
+  describe('select all button', () => {
+    test('is shown', async () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+      store.dispatch(setTerm('201931'));
+      store.dispatch<any>(updateCourseCard(0, makeCourseCard({})));
+
+      // act
+      const { getByText } = render(
+        <Provider store={store}><SectionSelect id={0} /></Provider>,
+      );
+
+      // assert
+      expect(await getByText('SELECT ALL')).toBeInTheDocument();
+    });
+
+    describe('is checked', () => {
+      test('when all sections are selected manually', async () => {
+        // arrange
+        const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+        store.dispatch(setTerm('201931'));
+        store.dispatch<any>(updateCourseCard(0, makeCourseCard(
+          { sectionNum: '201', id: 123456 }, { sectionNum: '202', id: 123457 },
+        )));
+        const { getByText, getAllByDisplayValue } = render(
+          <Provider store={store}><SectionSelect id={0} /></Provider>,
+        );
+
+        // act
+        fireEvent.click(getByText('201'));
+        fireEvent.click(getByText('202'));
+
+        // assert
+        expect(getAllByDisplayValue('allOn')).toHaveLength(1);
+      });
+    });
+
+    describe('is unchecked', () => {
+      test('by default', async () => {
+        // arrange
+        const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+        store.dispatch(setTerm('201931'));
+        store.dispatch<any>(updateCourseCard(0, makeCourseCard({})));
+
+        // act
+        const { getAllByDisplayValue } = render(
+          <Provider store={store}><SectionSelect id={0} /></Provider>,
+        );
+
+        // assert
+        expect(getAllByDisplayValue('allOff')).toHaveLength(1);
+      });
+
+      test('when an individual section is unchecked', async () => {
+        // arrange
+        const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+        store.dispatch(setTerm('201931'));
+        store.dispatch<any>(updateCourseCard(0, makeCourseCard(
+          { sectionNum: '201', id: 123456 }, { sectionNum: '202', id: 123457 },
+        )));
+        const { getByText, getAllByDisplayValue } = render(
+          <Provider store={store}><SectionSelect id={0} /></Provider>,
+        );
+
+        // act
+        fireEvent.click(getByText('201'));
+        fireEvent.click(getByText('202'));
+        fireEvent.click(getByText('201'));
+
+        // assert
+        expect(getAllByDisplayValue('allOff')).toHaveLength(1);
+      });
+    });
+
+    test('selects all sections', async () => {
```

Comment:
```suggestion
    test('when we select all sections', async () => {
```
the `describe` and their children should read like a sentence - so for this it'd be `select all is unchecked when we select all sections`

On that note - I believe this one needs test should be under the `is checked` block

Reply:
Sorry, this test is actually under the broad describe, so it would read `select all selects all sections`.
I changed it to `select all selects all sections when clicked` since writing that out felt weird.

The reason I didn't put it in the `is checked` block is because the `is checked` and `is unchecked` blocks were meant to be a reactive thing, while this test is more an active thing.

Reply:
Sounds good, that makes sense